### DISCUSSION TITLE
Is setting strict true not advisable for typescript when creating mantine components?

### DISCUSSION THREAD

**Discussion by vikigenius** [mantinedev/mantine#2747] [Q&A]
Looking at the codebase there is a typical pattern

https://github.com/mantinedev/mantine/blob/37a6b4847f355b58a86d9618adfc84e2f801e2fa/src/mantine-core/src/Chip/Chip.tsx#L67-L89

Then it is used in styling like so

https://github.com/mantinedev/mantine/blob/37a6b4847f355b58a86d9618adfc84e2f801e2fa/src/mantine-core/src/Chip/Chip.tsx#L94-L97

Here `radius`, `color` and `size` will throw errors about undefined being unassignable.

For example `Type 'MantineNumberSize | undefined' is not assignable to type 'MantineNumberSize'.`

Is this something to be concerned about ?If I am creating my own component, what's the right way to avoid this?

Should the style definition be changed to allow null/undefined ?

https://github.com/mantinedev/mantine/blob/37a6b4847f355b58a86d9618adfc84e2f801e2fa/src/mantine-core/src/Chip/Chip.styles.ts#L35-L39

**Comment by rtivital** [comment]
We do not use strict mode while in Mantine development, you can use strict mode in your application if you want to

**Reply by vikigenius** [reply]
Thanks for the response. If for example I was just replicating the Chip component as shown above in my codebase with strict on. What would be the ideal way to avoid the `Type 'MantineNumberSize | undefined' is not assignable to type 'MantineNumberSize'.` error for example ?

**Reply by rtivital** [reply]
Make sure that the value that you are passing is not undefined or make the function accept undefined as a value.

**Reply by vikigenius** [reply]
@rtivital So to avoid the undefined unassignable error in strict mode I tried casting the below code

https://github.com/mantinedev/mantine/blob/37a6b4847f355b58a86d9618adfc84e2f801e2fa/src/mantine-core/src/Chip/Chip.tsx#L94-L97

To this

```ts
  const { classes, cx } = useStyles(
    { radius: radius as MantineNumberSize, size: size as MantineSize, color: color as MantineColor },
    { classNames, styles, unstyled, name: 'Chip' } // The styles here throws an error now
  );
```
But this throws

```ts
src/components/Chip/Chip.tsx:98:19 - error TS2322: Type 'Styles<"root" | "label" | "iconWrapper" | "input" | "checkIcon", ChipStylesParams> | undefined' is not assignable to type 'Partial<Record<"root" | "label" | "iconWrapper" | "input" | "checkIcon", CSSObject>> | ((theme: MantineTheme, params: Record<string, any>) => Partial<...>) | undefined'.
  Type '(theme: MantineTheme, params: ChipStylesParams) => Partial<Record<"root" | "label" | "iconWrapper" | "input" | "checkIcon", CSSObject>>' is not assignable to type 'Partial<Record<"root" | "label" | "iconWrapper" | "input" | "checkIcon", CSSObject>> | ((theme: MantineTheme, params: Record<string, any>) => Partial<...>) | undefined'.
    Type '(theme: MantineTheme, params: ChipStylesParams) => Partial<Record<"root" | "label" | "iconWrapper" | "input" | "checkIcon", CSSObject>>' is not assignable to type '(theme: MantineTheme, params: Record<string, any>) => Partial<Record<"root" | "label" | "iconWrapper" | "input" | "checkIcon", CSSObject>>'.
      Types of parameters 'params' and 'params' are incompatible.
        Type 'Record<string, any>' is missing the following properties from type 'ChipStylesParams': radius, size, color

98     { classNames, styles, unstyled, name: 'Chip' }
```

I am a bit unclear on why the second set of arguments is needed
`{ classNames, styles, unstyled, name: 'Chip' }`

So it is hard to understand what the error means
